bin/privatebin-admin

```diff
@@ -0,0 +1,184 @@
+        if ($conf->getKey('class', 'model') !== 'Filesystem') {
+            error('instance not using Filesystem storage, no directories to empty', 4);
+        }
+        $dir = $conf->getKey('dir', 'model_options');
+        passthru("find $dir -type d -empty -delete", $code);
+        exit($code);
+    }
+
+    $class = 'PrivateBin\\Data\\' . $conf->getKey('class', 'model');
+    $store = new $class($conf->getSection('model_options'));
+
+    if (($pasteid = option('d', 'delete')) !== null) {
+        if (!Paste::isValidId($pasteid)) {
+            error('given ID is not a valid paste ID (16 hexadecimal digits)', 5);
+        }
+        if (!$store->exists($pasteid)) {
+            error('given ID does not exist, has expired or was already deleted', 6);
+        }
+        $store->delete($pasteid);
+        if ($store->exists($pasteid)) {
+            error('paste ID exists after deletion, permission problem?', 7);
+        }
+        exit("paste $pasteid successfully deleted" . PHP_EOL);
+    }
+
+    if (option('p', 'purge') !== null) {
+        $store->purge(PHP_INT_MAX);
+        exit('purging of expired pastes concluded' . PHP_EOL);
+    }
+
+    if (option('s', 'statistics') !== null) {
+        $counters = array(
+            'burn'          => 0,
+            'discussion'    => 0,
+            'expired'       => 0,
+            'md'            => 0,
+            'percent'       => 1,
+            'plain'         => 0,
+            'progress'      => 0,
+            'syntax'        => 0,
+            'total'         => 0,
+            'unknown'       => 0,
+        );
+        $time = time();
+        $ids = $store->getAllPastes();
+        $counters['total'] = count($ids);
+        $dots = $counters['total'] < 100 ? 10 : (
+            $counters['total'] < 1000 ? 50 : 100
+        );
+        $percentages = $counters['total'] < 100 ? 0 : (
+            $counters['total'] < 1000 ? 4 : 10
+        );
+        echo "Total:\t\t\t${counters['total']}", PHP_EOL;
+        foreach ($ids as $pasteid) {
+            $paste = $store->read($pasteid);
+            ++$counters['progress'];
+            if (
+                array_key_exists('expire_date', $paste['meta']) &&
+                $paste['meta']['expire_date'] < $time
+            ) {
+                ++$counters['expired'];
+            }
+            if (array_key_exists('adata', $paste)) {
+                $format = $paste['adata'][1];
+                $discussion = $paste['adata'][2];
+                $burn = $paste['adata'][3];
+            } else {
+                $format = array_key_exists('formatter', $paste['meta']) ? $paste['meta']['formatter'] : 'plaintext';
+                $discussion = array_key_exists('opendiscussion', $paste['meta']) ? $paste['meta']['opendiscussion'] : false;
+                $burn = array_key_exists('burnafterreading', $paste['meta']) ? $paste['meta']['burnafterreading'] : false;
+            }
+            if ($format === 'plaintext') {
```

Comment:
That could be refactored into a private method or so to get the correct value.

Reply:
Did extract those as recommended and wrap it into a more OOP style incl. docblocks as we do elsewhere. Also renamed it, as it is part of privatebin, so name seemed a bit redundant.